src/app/storage.js

```diff
@@ -28,12 +33,16 @@ function Storage () {
 
   this.remove = function (name) {
     window.localStorage.removeItem('sol:' + name)
+    return true
   }
 
   this.rename = function (originalName, newName) {
     var content = this.get(originalName)
-    this.set(newName, content)
+    if (!this.set(newName, content)) {
```

Comment:
I don't think we should follow posix here. What is the use-case of overwriting by renaming? I know that we do not make atomic promises here, but I still think that it is better to fail in this case. After all, this is an IDE and not an operating system. If you rename a file you are currently working on, you want to change its name and not another file to disappear at the same time.